nbconvert/nbconvertapp.py

```diff
@@ -30,6 +31,36 @@
 #Classes and functions
 #-----------------------------------------------------------------------------
 
+class CaselessStrEnumOrDottedObjectName(CaselessStrEnum, DottedObjectName):
```

Comment:
I think rather than defining a fiddly new traitlet type to hold this, I'd just drop it down to a `Unicode` trait, and leave checking whether it refers to a real exporter until we try to use it.